### PR TITLE
FreeBSD support for Puppet module

### PR DESCRIPTION
**Pull request by olofhellqvist**
Hi,
Great module! I have implemented FreeBSD support for it.

Best Regards Olof Hellqvist

### CODE REVIEW

File: manifests/service.pp [hunk 1]
```diff
@@ -4,4 +4,12 @@
...
     provider => $filebeat::service_provider,
   }
+  
+  if $filebeat::service_enable and $::kernel == 'FreeBSD' {
+    file_line {'/etc/rc.conf: filebeat_enable="YES"':
+      ensure => present,
+      path   => '/etc/rc.conf',
+      line   => 'filebeat_enable="YES"',
+    }
+  }
```
Comment (pcfens) [inline manifests/service.pp]: I think that the service enable piece is handled automatically by the BSD `service` resource provider. If true, we can leave this piece out.

File: manifests/prospector.pp [hunk 2]
```diff
@@ -31,6 +31,17 @@
   }
 
   case $::kernel {
+    'FreeBSD' : {
+      file { "filebeat-${name}":
+        ensure  => $ensure,
+        path    => "${filebeat::config_dir}/${name}.yml",
+        owner   => 'root',
+        group   => 'wheel',
```
Comment (pcfens) [inline manifests/prospector.pp]: It looks like the only difference here is the group. Does it make more sense to parameterize the group or have a separate resource?

File: manifests/install/freebsd.pp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class filebeat::install::freebsd {
+  package {'filebeat':
+    ensure => $filebeat::package_ensure,
+  }
+  file { $filebeat::directory:
```
Comment (pcfens) [inline manifests/install/freebsd.pp]: Is this to create the base configuration directory?

File: manifests/init.pp [hunk 4]
```diff
@@ -59,6 +60,7 @@
   $config_dir_mode      = $filebeat::params::config_dir_mode,
   $config_dir           = $filebeat::params::config_dir,
   $config_file_mode     = $filebeat::params::config_file_mode,
+  $directory            = $filebeat::params::directory,
```
Comment (pcfens) [inline manifests/init.pp]: It looks like `config_dir` might have been a bad parameter name to use for the prospector configuration directory, and would have been better served here.

Before I start looking for ways to change this around without breaking anything, is this only being used to create the configuration base directory?

**Comment by pcfens** [comment]
Thanks so much for the PR to add BSD support.

I've attached some questions to the diff to help me understand everything a little better.

I'd like to add some tests around all of this too just to make sure nothing gets missed, but it may make more sense to write those after parameter names are all settled out.

File: manifests/service.pp (hunk 1 above)
Comment (olofhellqvist) [inline manifests/service.pp]: I though so myself but the FreeBSD provider doesn’t handle it

File: manifests/prospector.pp (hunk 2 above)
Comment (olofhellqvist) [inline manifests/prospector.pp]: Having a parameter would make more sense, so you can override it as well

File: manifests/install/freebsd.pp (hunk 3 above)
Comment (olofhellqvist) [inline manifests/install/freebsd.pp]: Yes the directory is not created by default.

**Comment by pcfens** [comment]
I'm going to go ahead and close this since it looks like we got support in #130. Comment or create an issue if I've missed something.

**Pull request closed without merging**